staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -84,36 +85,31 @@ func (a statusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.O
 
 // ValidateUpdate is the default update validation for an end user updating status.
 func (a statusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
-	var errs field.ErrorList
-	errs = append(errs, a.customResourceStrategy.validator.ValidateStatusUpdate(ctx, obj, old, a.scale)...)
-
 	uNew, ok := obj.(*unstructured.Unstructured)
 	if !ok {
-		return errs
+		return field.ErrorList{field.Invalid(field.NewPath(""), obj, fmt.Sprintf("has type %T. Must be a pointer to an Unstructured type", obj))}
 	}
 	uOld, ok := old.(*unstructured.Unstructured)
-	var oldObject map[string]interface{}
 	if !ok {
-		oldObject = nil
```

Comment:
This is a functional change, passing with non-unstructured objects before, now it is an error. Wondering whether we anything in mind here. Scale subresource comes to mind. Is that the same strategy? Probably not.

Reply:
This was added relatively recently and seemingly as a defensive measure: https://github.com/kubernetes/kubernetes/pull/108073/files#diff-0508ef6d93f19e18a1b44e6e02ae3ff4cac9cc8065ff5c41e2bea4b9eeb6ee0bR95

Gladly would add logic to allow untypted `nil`, but curious if we actually want to?

Reply:
Also note that this error return was moved from `ValidateStatusUpdate`, so passing untyped nil was always an error.

Reply:
Right. In my own words: the `customResourceValidator` always error'ed on non-unstructured. So everywhere (and that is everywhere in this PR) where we called the `customResourceValidator` and then did conditional logic on unstructured'ness, we can return a hard error directly without regressing.